app/mails/tools.py

```diff
@@ -24,9 +25,14 @@ def get_delay_days_from_email_address(email_address):
     :rtype: integer
     """
     try:
-        match = re.findall(r"^(\d+)([dmw])", email_address)[0]
-        multiplicator = settings.EMAIL_SUFFIX_TO_DAY[match[1]]
-        delay = int(match[0]) * int(multiplicator)
+        if re.match(r"^(\d+)([dmw])", email_address):
+            match = re.findall(r"^(\d+)([dmw])", email_address)[0]
+            multiplicator = settings.EMAIL_SUFFIX_TO_DAY[match[1]]
+            delay = int(match[0]) * int(multiplicator)
+        delay_datetime = dateparser.parse(
+            email_address.split("@")[0], settings=settings.DATEPARSER_SETTINGS
```

Comment:
You'll still need to strip out the anti-spam key here I think. Not just the before-the-@ bit from the address